src/framework/circuit.hpp

```diff
@@ -252,7 +252,8 @@ void Circuit::add_op_metadata(const Op& op) {
 void Circuit::set_params(bool truncation) {
   // Clear current circuit metadata  
   reset_metadata();
-  
+  if (ops.empty()) return;
+
```

Comment:
This early return means that the check a few lines down `if (!ops.empty())` is now guaranteed to be true.